src/main/java/de/gebit/plugins/autoconfig/actions/RunAutoconfigAction.java

```diff
@@ -23,6 +30,31 @@ public void actionPerformed(@NotNull AnActionEvent e) {
 			configurationLoaderService.resetConfigurationCache();
 		}
 
-		new AutoconfigStartup().runAutoconfig(project);
+		// save currently open and unsaved config files
+		saveUnsavedAutoconfigs();
+
+		// perform changes
+		List<String> changedSettings = new AutoconfigStartup().runAutoconfig(project);
+
+		// notify user if no changes were necessary
+		if (changedSettings == null || changedSettings.isEmpty()) {
+			Notifications.showInfo("No settings have been changed.", project);
+		}
+	}
+
+	private void saveUnsavedAutoconfigs() {
+		FileDocumentManager fileDocumentManager = FileDocumentManager.getInstance();
+		Arrays.stream(fileDocumentManager.getUnsavedDocuments())
+				.filter(d -> isAutoconfigFile(fileDocumentManager.getFile(d)))
+				.forEach(fileDocumentManager::saveDocument);
+	}
+
+	private boolean isAutoconfigFile(@Nullable VirtualFile document) {
+		if (document == null) {
+			return false;
+		}
+		String[] components = document.getUrl().split("/");
+		String lastComponent = components[components.length - 1];
+		return lastComponent.startsWith("autoconfig") && lastComponent.endsWith(".yaml");
```

Comment:
```suggestion
		String fileName = document.getName()
		return fileName.startsWith("autoconfig") && fileName.endsWith(".yaml");
```
Je nach System könnten unterschiedliche File-Separatoren verwendet werden. Der Dateiname ist möglicherweise nicht in allen Fällen durch '/' vom vorherigen Pfad getrennt.

Reply:
Zum Glück ist das eine URL, da kann ich mich drauf verlassen denke ich ;-)